DM-24786: Add string length constraint checks to sqlite and fix component length

timj opened the pull request.

timj committed:
Remove old debug print

timj committed:
Increase length of component field

16 was too short for "TransmissionCurve"

timj committed:
Add explicit length constraint for sqlite string columns

sqlite doesn't care how long strings are but other databases
do. This leads to problems when we develop on sqlite but deploy
on postgres.

TallJimbo reviewed (approved):
Looks good!  It's a bit of a pity we had to add a new hook to the `Database` base class, but I don't see a good way around it right now.

timj committed:
Ensure abstract filter length matches physical length

The problem we have is for non-standard filters we have
decided that the abstract filter name should be the
physical filter and therefore the length can be the same.
